src/csf_prf/engines/ENCReaderEngine.py

```diff
@@ -797,12 +801,28 @@ def unapproved(self, geom_type: str, properties: dict[str]) -> bool:
         """Check to ignore unapproved feature types"""
 
         objl_name = CLASS_CODES.get(int(properties['OBJL']), CLASS_CODES['OTHER'])[0]
+
+        if objl_name in self.get_unapproved_names():
+            return self.unapproved_subcategory(geom_type, objl_name, properties)
+
         unapproved_features = self.feature_lookup[geom_type]
         unapproved = False
-        for feature in unapproved_features:
-            if feature == objl_name:
-                unapproved = True
-                break
+        if objl_name in unapproved_features:
+            unapproved = True
```

Comment:
I have no clue why I was looping through values before just to see if one matched.